lib/money/currency.rb

```diff
@@ -32,6 +34,19 @@ class UnknownCurrency < ArgumentError; end
 
     class << self
 
+      def new(id)
+        id = id.to_s.downcase
+        unless stringified_keys.include?(id)
+          raise UnknownCurrency, "Unknown currency '#{id}'"
+        end
+
+        instances[id] || super
```

Comment:
@printercu Good mention about passing id to super, but i'm wondering why are the tests still green when its called without args.